### DISCUSSION TITLE
Is there a file upload function？

### DISCUSSION THREAD

**Discussion by AnkoGo123** [jchristn/HttpServerLite#6] [General]
Is there a file upload function？

**Comment by jchristn** [comment]
A file upload is just a data stream attached to an HTTP request using a method/verb that supports a payload (such as POST, PUT, etc).  You can absolutely either A) create routes (using the appropriate HTTP method/verb) or B) use the default handler to access the underlying data stream from a client's request.

**Reply by jchristn** [reply]
For example:
```csharp

[StaticRoute(HttpMethod.POST, "/upload")]
public static async Task MyUploadRoute(HttpContext ctx)
{
  Console.WriteLine("Received " + ctx.Request.ContentLength + " bytes from client");
  // your data is in ctx.Request.Data
  // do something, and send a response
}
```
